processor/groupbyattrsprocessor/attribute_groups.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbyattrsprocessor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// attributeEntry keeps a grouping attributes and matching records
+type attributeEntry struct {
+	attrs         pdata.AttributeMap
+	resourceSpans pdata.ResourceSpans
+	resourceLogs  pdata.ResourceLogs
+}
+
+func instrumentationLibrariesEqual(il1, il2 pdata.InstrumentationLibrary) bool {
```

Comment:
I'm not right - operator == compares the underlying pointers to 2 strings. In this case, what do you think about making`Equal` function on the struct instead of this?

Reply:
Do you mean adding `Equal` to the internal API of the models?

Reply:
Yes, exactly. @bogdandrutu what do you think about that?